src/WLR/Types/Pointer.hsc

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE EmptyDataDeriving #-}
+module WLR.Types.Pointer (
+    WLR_pointer(..)
+    ) where
+
+#define WLR_USE_UNSTABLE
+#include <wlr/types/wlr_pointer.h>
+
+import Foreign.Ptr (
+    Ptr
+    )
+import Foreign.C.String (
+    CString
+    )
+import Foreign.Storable (
+    Storable(..)
+    )
+
+import WLR.Types.InputDevice (
+    WLR_input_device
+    )
+import WL.ServerCore (
+    WL_signal
+    )
+
+data {-# CTYPE "wlr/types/wlr_pointer.h" "struct wlr_pointer_impl" #-} WLR_pointer_impl
+    deriving (Show)
+
+data {-# CTYPE "wlr/types/wlr_pointer.h" "struct wlr_pointer" #-} WLR_pointer
+    = WLR_pointer
+    { wlr_pointer_base :: Ptr WLR_input_device
```

Comment:
Should just be `WLR_input_device` instead of `Ptr WLR_input_device`, right?

Reply:
Correct.